### PR TITLE
js_sliding-tree-DOM: Cypress tests added

### PR DESCRIPTION
**Pull request by iBrianWarner**

### CODE REVIEW

File: cypress/integration/js_sliding_tree.spec.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
+/* eslint-disable strict */
```
Comment (YuriiHoliuk) [inline cypress/integration/js_sliding_tree.spec.js:1]: use camelCase for fileNames(`jsSlidingTree.spec.js`)

File: cypress/integration/js_sliding_tree.spec.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+/* eslint-disable max-len */
+describe(('Tests for sliding tree.'), () => {
```
Comment (YuriiHoliuk) [inline cypress/integration/js_sliding_tree.spec.js:3]:
```suggestion
describe('Sliding tree.', () => {
```

File: cypress/integration/js_sliding_tree.spec.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  before(() => {
+    cy.visit('/');
+  });
+
+  it('Fruits block.', () => {
```
Comment (YuriiHoliuk) [inline cypress/integration/js_sliding_tree.spec.js:8]: it message should start from the `should` word or similar

File: cypress/integration/js_sliding_tree.spec.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    cy.contains('Fruit').click();
+
+    cy.get('body').then(($body) => {
+      if ($body.find('[hidden]').length > 0) {
+        cy.get('.tree > :nth-child(1) > ul').should('have.attr', 'hidden');
+      } else {
+        cy.get('.tree > :nth-child(1) > ul').should('have.attr', 'style', 'display: none;');
+      };
+    });
+    cy.contains('Fruit').click();
```
Comment (YuriiHoliuk) [inline cypress/integration/js_sliding_tree.spec.js:40]: too many actions in a single test

File: cypress/integration/js_sliding_tree.spec.js [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+  it('Fruits block.', () => {
+    cy.contains('Red').click();
+
+    cy.get('body').then(($body) => {
+      if ($body.find('[hidden]').length > 0) {
+        cy.get('.tree > :nth-child(1) > ul > :nth-child(1) > ul').should('have.attr', 'hidden');
+      } else {
+        cy.get('.tree > :nth-child(1) > ul > :nth-child(1) > ul').should('have.attr', 'style', 'display: none;');
+      };
```
Comment (YuriiHoliuk) [inline cypress/integration/js_sliding_tree.spec.js:16]: avoid conditional testing

File: cypress/integration/js_sliding_tree.spec.js [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+      if ($body.find('[hidden]').length > 0) {
+        cy.get('.tree > :nth-child(1) > ul').should('have.attr', 'hidden');
+      } else {
+        cy.get('.tree > :nth-child(1) > ul').should('have.attr', 'style', 'display: none;');
+      };
+    });
+    cy.contains('Fruit').click();    
+  });
+
+  it('Fish block.', () => {
```
Comment (YuriiHoliuk) [inline cypress/integration/js_sliding_tree.spec.js:43]: нам не потрібно тестувати блоки - потрібно тестувати функціонал

File: cypress/integration/js_sliding_tree.spec.js (hunk 1 above)
Comment (iBrianWarner) [reply cypress/integration/js_sliding_tree.spec.js]: перейменував тут і в інших задачах

File: cypress/integration/js_sliding_tree.spec.js (hunk 2 above)
Comment (iBrianWarner) [reply cypress/integration/js_sliding_tree.spec.js]: перейменував

File: cypress/integration/js_sliding_tree.spec.js (hunk 3 above)
Comment (iBrianWarner) [reply cypress/integration/js_sliding_tree.spec.js]: done

File: cypress/integration/js_sliding_tree.spec.js (hunk 4 above)
Comment (iBrianWarner) [reply cypress/integration/js_sliding_tree.spec.js]: розбив тести на ті, що перевіряють, що елемент схований та те, що він знову з'являється при кліку. але залишив прив'язку до блоків, ящетаю так буде легше

File: cypress/integration/js_sliding_tree.spec.js (hunk 5 above)
Comment (iBrianWarner) [reply cypress/integration/js_sliding_tree.spec.js]: прибрав conditional testing

File: README.md [hunk 7]
```diff
@@ -1,7 +1,9 @@
 1. Replace `<your_account>` with your Github username in the link
     - [DEMO LINK](https://<your_account>.github.io/js_sliding-tree-DOM/)
 2. Follow [this instructions](https://mate-academy.github.io/layout_task-guideline/)
-    - There are no tests for this task so use `npm run lint` command instead of `npm test` 
+    - Run `npm run test` command to test your code;
+    - Run `npm run test:only -- -n` to run fast test ignoring linter;
+    - Run `npm run test:only -- -l` to run fast test with additional info in console ignoring linter.
```
Comment (roman-mate) [inline README.md:6]: plz try to fix grammar mistakes in the whole README doc.

File: cypress/integration/slidingTreeDom.spec.js [hunk 8]
```diff
@@ -0,0 +1,53 @@
+'use strict';
+
+describe(('Sliding tree'), () => {
+  beforeEach(() => {
+    cy.visit('/');
+    cy.get('.tree > :nth-child(1) > ul').as('fruits');
+    cy.get('.tree > :nth-child(2) > ul').as('fishes');
```
Comment (roman-mate) [inline cypress/integration/slidingTreeDom.spec.js:7]: are there better selectors?

File: cypress/integration/slidingTreeDom.spec.js [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+
+    cy.contains('Yellow').click();
+    cy.get('@fruits').find(':nth-child(2) > ul').should('not.be.visible');
+
+    cy.contains('Red').click();
+    cy.get('@fruits').find(':nth-child(1) > ul').should('be.visible');
+
+    cy.contains('Yellow').click();
+    cy.get('@fruits').find(':nth-child(2) > ul').should('be.visible');
+  });
```
Comment (roman-mate) [inline cypress/integration/slidingTreeDom.spec.js:30]: why do we need to click on both read and yellow? What are we testing here?

File: cypress/integration/slidingTreeDom.spec.js [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+    cy.get('@fishes').find(':nth-child(1) > ul').should('not.be.visible');
+
+    cy.contains('Freshwater fish').click();
+    cy.get('@fishes').find(':nth-child(2) > ul').should('not.be.visible');
+
+    cy.contains('Ocean fishes').click();
+    cy.get('@fishes').find(':nth-child(1) > ul').should('be.visible');
+
+    cy.contains('Freshwater fish').click();
+    cy.get('@fishes').find(':nth-child(2) > ul').should('be.visible');
```
Comment (roman-mate) [inline cypress/integration/slidingTreeDom.spec.js:51]: what's the minimal possible test to cover scenario in this test?

**Review by roman-mate — CHANGES_REQUESTED** [review]

File: cypress/integration/slidingTreeDom.spec.js (hunk 10 above)
Comment (iBrianWarner) [reply cypress/integration/slidingTreeDom.spec.js:51]: Click on one of the elements in the first level and one of the elements in the second. But I've decided to click on all of the elements.

File: cypress/integration/slidingTreeDom.spec.js [hunk 11]
```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+Cypress.Commands.add('hideTree', (selector) => {
```
Comment (roman-mate) [inline cypress/integration/slidingTreeDom.spec.js:3]: we are not just hiding. We are hiding, then showing.

File: cypress/integration/slidingTreeDom.spec.js [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+  cy.get('ul').contains(selector).click();
+
+  cy.get('ul').contains(selector).parent().should(($ul) => {
+    const ul = $ul.find('ul').is(':hidden');
```
Comment (roman-mate) [inline cypress/integration/slidingTreeDom.spec.js:7]: we expect the value of variable ul to be equal to true or false. From the variable name it's not clear that variable contains boolena. Need a better name.

File: cypress/integration/slidingTreeDom.spec.js [hunk 13]
```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+Cypress.Commands.add('hideTree', (selector) => {
+  cy.get('ul').contains(selector).click();
+
+  cy.get('ul').contains(selector).parent().should(($ul) => {
```
Comment (roman-mate) [inline cypress/integration/slidingTreeDom.spec.js:6]: would something like below work
cy.get('ul').contains(selector).children().should('not.be.visible')?

File: cypress/integration/slidingTreeDom.spec.js (hunk 8 above)
Comment (iBrianWarner) [reply cypress/integration/slidingTreeDom.spec.js]: all selectors were changed

File: cypress/integration/slidingTreeDom.spec.js (hunk 13 above)
Comment (iBrianWarner) [reply cypress/integration/slidingTreeDom.spec.js:6]: nope, it wouldn't. 
cy.get('ul').contains(selector) or cy.contains(selector)  - selects <span> element without children.

File: cypress/integration/slidingTreeDom.spec.js (hunk 12 above)
Comment (iBrianWarner) [reply cypress/integration/slidingTreeDom.spec.js:7]: changed to isVisible/isHidden

File: cypress/integration/slidingTreeDom.spec.js (hunk 11 above)
Comment (iBrianWarner) [reply cypress/integration/slidingTreeDom.spec.js:3]: slideTree would be better?

**Review by roman-mate — APPROVED** [review]
Great job!

**Pull request merged**